src/command.rs

```diff
@@ -536,9 +536,11 @@ impl CliConfiguration<Self> for RelayChainCli {
 	}
 
 	fn base_path(&self) -> Result<Option<BasePath>> {
-		self.shared_params()
+		Ok(self
+			.shared_params()
 			.base_path()
-			.or_else(|_| Ok(self.base_path.clone().map(Into::into)))
+			.unwrap_or_else(|_| self.base_path.clone().map(Into::into))
+			.or_else(|| self.base_path.clone().map(Into::into)))
```

Comment:
This means in hierarchical order:

1. Take the path directly provided as relay chain argument
2. If not present use the one `RelayChainCli` was given as a field
3. ???

Why the last `or_else` clause()? If this intends to call the logic of the substrate core cli, shouldnt' this be `self.base_path()`

Reply:
This was my starting point for this:

```
match self.shared_params().base_path() {
			Ok(base_path_opt) => match base_path_opt {
				Some(base_path) => Ok(base_path.into()),
				None => Ok(self.base_path.clone().map(Into::into)),
			},
			Err(_) => Ok(self.base_path.clone().map(Into::into)),
		}
```

Reply:
Updated to a cleaner version, as per our Slack conversation. Thank you! :heart: